File: scala-spark-master/scala-spark-master/src/main/scala/com/com/simplilearn/bigdata/spark/TimeUtils.scala
package com.com.simplilearn.bigdata.spark

import java.sql.Timestamp

import java.util.Calendar

object TimeUtils {
  private val monthName = Array("January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December")
  private val quaters = Array("1st Quarter", "2nd Quarter", "3rd Quarter", "4th Quarter")

  def getMonth(timestamp: Timestamp): Integer = {
    val time = timestamp.getTime
    val cal = Calendar.getInstance
    cal.setTimeInMillis(time)
    cal.get(Calendar.MONTH)
  }

  def getYear(timestamp: Timestamp): Integer = {
    val time = timestamp.getTime
    val cal = Calendar.getInstance
    cal.setTimeInMillis(time)
    cal.get(Calendar.YEAR)
  }

  def numberToMonthName(number: Int): String = monthName(number)

  def numberToQuarter(number: Int): String = quaters(number - 1)

  def getQuater(timestamp: Timestamp): Integer = {
    val time = timestamp.getTime
    val cal = Calendar.getInstance
    cal.setTimeInMillis(time)
    (cal.get(Calendar.MONTH) / 3) + 1
  }
}

File: spark-retail-transaction-analysis-master/src/main/java/com/simplilearn/bigdata/spark/TimeUtils.java
package com.simplilearn.bigdata.spark;

import java.sql.Timestamp;
import java.util.Calendar;

public class TimeUtils {

    private static String[] monthName = {"January", "February",
            "March", "April", "May", "June", "July",
            "August", "September", "October", "November",
            "December"};

    private static String[] quaters = {"1st Quarter", "2nd Quarter",
            "3rd Quarter", "4th Quarter"};

    public static Integer getMonth(Timestamp timestamp) {
        long time = timestamp.getTime();
        Calendar cal = Calendar.getInstance();
        cal.setTimeInMillis(time);
        return cal.get(Calendar.MONTH);
    }

    public static Integer getYear(Timestamp timestamp) {
        long time = timestamp.getTime();
        Calendar cal = Calendar.getInstance();
        cal.setTimeInMillis(time);
        return cal.get(Calendar.YEAR);
    }

    public static String numberToMonthName(int number) {
        return monthName[number];
    }

    public static String numberToQuarter(int number) {
        return quaters[number-1];
    }

    public static Integer getQuater(Timestamp timestamp) {
        long time = timestamp.getTime();
        Calendar cal = Calendar.getInstance();
        cal.setTimeInMillis(time);
        return (cal.get(Calendar.MONTH) / 3) + 1;
    }
}
